theia-ide/theia
# New `@theia/example-hybrid` application

This PR adds support for an electron client to connect to an hybrid backend.

Currently done:
- [x] New `@theia/example-hybrid` package that can serve both web browsers and electron apps.
- [x] Modified the generators to autogen `@theia/example-hybrid`.
- [x] New Electron-specific commands to choose a remote host to connect to.
  - [x] Basic host connection history.
  - [x] Host polling to see status (online/offline).
- [x] Some issues when actually loading remote content in Electron.
  -  It seems to come from the AMD loader importing the monaco editor.
- [ ] Security options are not yet set to accommodate for remote content loading
  - Such as disabling node integration and such...

---

How to test this PR:

- Use `node v8.2.1`, it should be the same as the one used by Electron, so that you don't have to rebuild native modules when starting `@theia/example-electron` and `@theia/example-hybrid`
```sh
nvm install 8.2.1
nvm alias electron 8.2.1
nvm alias default electron
nvm use electron
```
- Start both `@theia/example-electron` and `@theia/example-hybrid` so that you have a server to connect to.
- Open the command palette (<kbd>F1</kbd>) and use `Remote: Connect to a server`.
